examples/technical_blogs/tartan_blog/build_and_run_networks.py

```diff
@@ -0,0 +1,156 @@
+
+# Build SYCL-DNN
+def build_sycl_dnn():
+    script_directory = os.getcwd()
+    sycl_dnn_build_dir = os.path.join(script_directory, "build_sycl_dnn")
+    
+    create_directory(sycl_dnn_build_dir)
+    os.chdir(sycl_dnn_build_dir)
+    
+    cmake_sycl_dnn_command = [
+        "cmake",
+        f"{SYCL_DNN_source}",
+        "-GNinja",
+        f"-DCMAKE_CXX_COMPILER={os.environ['CMAKE_CXX_COMPILER']}",
+        "-DSNN_BUILD_BENCHMARKS=OFF",
+        "-DSNN_BENCH_SYCLBLAS=OFF",
+        "-DSNN_BUILD_DOCUMENTATION=OFF",
+        f"-DSyclBLAS_DIR='{os.path.join(script_directory, 'build_portBlas')}'"
+    ]
+
+    run_cmake_and_ninja(cmake_sycl_dnn_command)
+    os.chdir("..")
+
+
+def download_and_convert_model(file_url, destination_file, h5_to_bin_script):
+    try:
+        response = requests.get(file_url)
+        response.raise_for_status()
+        with open(destination_file, "wb") as file:
+            file.write(response.content)
+        print(f"Downloaded '{destination_file}' successfully.")
+        
+        subprocess.run(["python", h5_to_bin_script, destination_file], check=True)
+        print(f"Converted '{destination_file}' to binary successfully.")
+    except requests.exceptions.RequestException as e:
+        print(f"Error downloading file: {e}")
+    except subprocess.CalledProcessError as e:
+        print(f"Error running h5toBin.py: {e}")
+
+
+def prepare_image(img_script_path, input_image):
+    try:
+        subprocess.run(["python", img_script_path, input_image], check=True)
+        print(f"Image prepared successfully.")
+    except subprocess.CalledProcessError as e:
+        print(f"Error executing the script: {e}")
+
+# Test on images
+def test_image(model_script, data_dir, output_file):
+    command_args = [
+        model_script,
+        data_dir + "/",
+        output_file
+    ]
+    try:
+        subprocess.run(command_args, check=True)
+        print(f"Image tested successfully with {os.path.basename(model_script)}.")
+    except subprocess.CalledProcessError as e:
+        print(f"Error executing the command: {e}")
+
+
+if __name__ == "__main__":
+    build_portBLAS()
+    build_sycl_dnn()
+    sycl_dnn_build_dir = os.path.join(os.getcwd(), "build_sycl_dnn")
+
+    # Set the environment variables
+    os.environ["CA_HAL_DEBUG"] = "1"
+    os.environ["CA_PROFILE_LEVEL"] = "3"
+    os.environ["OCL_ICD_FILENAMES"] = f"{OCK_libCL_path}"
+    os.environ["ONEAPI_DEVICE_SELECTOR"] = "opencl::acc"
+    os.environ["SYCL_CONFIG_FILE_NAME"] = "null.cfg"
+
+    # VGG16
+    vgg_dir = "vdata"
+    create_directory(vgg_dir)
+    os.chdir(vgg_dir)
+    file_url = "https://storage.googleapis.com/tensorflow/keras-applications/vgg16/vgg16_weights_tf_dim_ordering_tf_kernels.h5"
+    destination_file = "vgg16_weights_tf_dim_ordering_tf_kernels.h5"
+    h5_to_bin_script = os.path.join(SYCL_DNN_source, "samples", "networks", "vgg", "h5toBin.py")
+    
+    download_and_convert_model(file_url, destination_file, h5_to_bin_script)
```

Comment:
I wonder if google's APIs would block a user after requesting to download this file over and over in a short period of time? E.g., if a developer repeatedly runs this scripts but keeps hitting errors during development.

The only way around that would be to check if the destination file already exists and either a) trust that it's correct or b) use some kind of hash to check whether the contents are identical?

Reply:
I can look into it, but for now I am trusting that if the file exisits already then its the correct one.